### DISCUSSION TITLE
How could I access the request URL in a composable?

### DISCUSSION THREAD

**Discussion by martinszeltins** [nuxt/content#2238] [General]
I need to know the request URL in a composable with SSR. Since window object is not available in SSR mode, is there a way I could get access to the request URL in my component or composable? I believe Nuxt nitro server should have access to the request when it is rendering the components on the server, right?

Basically, the reason I need this is so that I could change the API endpoint based on if the user is using `localhost` or `https://....` - but how do I get access to this?

**Comment by nobkd** [comment]
Maybe the [`useRoute()` composable](https://nuxt.com/docs/api/composables/use-route) is what you're looking for?
Not sure if this is the way to go for, or if a [server middleware](https://nuxt.com/docs/guide/directory-structure/server#server-middleware) or so is better

**Comment by martinszeltins** [comment]
Ahh, `useRequestURL()` was what I was looking for. And `useRequestHeaders()` gives me all the headers.
